ui/cypress/e2e/tasks.test.ts

```diff
@@ -257,6 +257,112 @@ from(bucket: "${name}")
+      cy.getByTestID('task-card')
+        .should('have.length', 1)
+        .and('contain', firstTask)
+
+      cy.getByTestID('add-resource-dropdown--button').click()
+      cy.getByTestID('add-resource-dropdown--new').click()
+      cy.getByInputName('name').type(secondTask)
+      cy.getByTestID('task-form-schedule-input').type(interval)
+      cy.getByTestID('task-form-offset-input').type(offset)
+      cy.get<Bucket>('@bucket').then(bucket => {
+        cy.getByTestID('flux-editor').within(() => {
+          cy.get('textarea').type(flux(bucket), {force: true})
+        })
+      })
+      cy.contains('Save').click()
+      cy.getByTestID('task-card')
+        .should('have.length', 2)
+        .and('contain', firstTask)
+        .and('contain', secondTask)
+      cy.getByTestID('task-card--name')
+        .contains(firstTask)
+        .click()
+    })
+
+    it('when navigating using the navbar', () => {
+      // verify that the previously input data exists
+      cy.getByInputValue(firstTask)
+      // navigate home
+      cy.get('div.cf-nav--item.active').click()
+      // click on the second task
+      cy.getByTestID('task-card--name')
+        .contains(secondTask)
+        .click()
+      // verify that it is the correct data
+      cy.getByInputValue(secondTask)
+      cy.get('div.cf-nav--item.active').click()
+      // navigate back to the first one to verify that the name is correct
+      cy.getByTestID('task-card--name')
+        .contains(firstTask)
+        .click()
+      cy.getByInputValue(firstTask)
+    })
+
+    it('when navigating using the cancel button', () => {
+      // verify that the previously input data exists
+      cy.getByInputValue(firstTask)
+      // navigate home
+      cy.getByTestID('task-cancel-btn').click()
+      // click on the second task
+      cy.getByTestID('task-card--name')
+        .contains(secondTask)
+        .click()
+      // verify that it is the correct data
+      cy.getByInputValue(secondTask)
+      cy.getByTestID('task-cancel-btn').click()
+      // navigate back to the first task again
+      cy.getByTestID('task-card--name')
+        .contains(firstTask)
+        .click()
+      cy.getByInputValue(firstTask)
+      cy.getByTestID('task-cancel-btn').click()
+    })
+
+    it('when navigating using the save button', () => {
+      // verify that the previously input data exists
+      cy.getByInputValue(firstTask)
+      // navigate home
+      cy.getByTestID('task-save-btn').click()
+      // click on the second task
+      cy.getByTestID('task-card--name')
+        .contains(secondTask)
+        .click()
+      // verify that it is the correct data
+      cy.getByInputValue(secondTask)
+      cy.getByTestID('task-save-btn').click()
+      // navigate back to the first task again
+      cy.getByTestID('task-card--name')
+        .contains(firstTask)
+        .click()
+      cy.getByInputValue(firstTask)
+      cy.getByTestID('task-save-btn').click()
+    })
+  })
```

Comment:
You could probably just have one "navigation" test which does all of these things in one test.  https://docs.cypress.io/guides/references/best-practices.html#Creating-%E2%80%9Ctiny%E2%80%9D-tests-with-a-single-assertion

Reply:
true, but the difference here is that we're testing how the state handles different click events. Even though the outcome is to navigate to the tasks page, we're testing to see whether the reducer acts differently depending on whether we're navigated via the navbar or the button clicks